### DISCUSSION TITLE
Up-to-date devtools guide?

### DISCUSSION THREAD

**Discussion by adgray** [phalcon/cphalcon#15493] [Q&A] [answered]
Are there any up-to-date, reasonably complete instructions on how to install the Phalcon devtools, that don't assume I understand Composer?

I tried the **Installing via Composer** steps at https://github.com/phalcon/phalcon-devtools, and they run without error, but don't create a `phalcon` binary anywhere I can find.

I've tried the **Installation via Git** steps on the same page, using the following specific steps: 

```
cd /usr/local
git clone https://github.com/phalcon/phalcon-devtools.git
ln -s /usr/local/phalcon-devtools/phalcon /usr/local/bin/phalcon
```

This works, but when I run the `phalcon` binary, I get this error:

`ERROR: Please run composer install
`

If I change into the `/usr/local/devtools` directory and run `curl -s http://getcomposer.org/installer | php` (surely that's how I install composer?) and then rerun the `phalcon` CLI tool I get the same error.

On a whim, I tried `composer.phar install` in the devtools directory. This takes ages, and manages to pull in over a gigabyte of code, even including symfony(!). And miraculously, now the `phalcon` CLI runs.

I don't use composer, and really don't need it for the project I'm working on, but surely the phalcon devtools don't actually require 1 gigabyte of dependencies, and *surely* they don't require symfony! How to I get the devtools, and just the devtools, installed?

**Comment by Jeckerson** [comment, marked as answer]
Hello

> This takes ages, and manages to pull in over a gigabyte of code

Could you send screenshot of `vendor/` folder size?
You can simply download `.phar` from latest release - https://github.com/phalcon/phalcon-devtools/releases and include inside your project.

**Reply by adgray** [reply]
Thanks - this eventually led me to figure out that I could download `phalcon.phar` from the releases area, rename it to phalcon, make it executable (chmod 755 phalcon), move it to /usr/local/bin, and it appears to work as described. I'm still wondering though why the official instructions can't just state it that simply. (I don't think I want it inside the project, although I'm not sure if not having it there will limit it in any way. I'm not even sure where in the project directory it would go!)

For what it's worth, here is a screenshot of the vendor folder size you asked for, once I ran `composer.phar install` in the devtools directory on my earlier attempt: 
<img width="264" alt="vendor" src="https://user-images.githubusercontent.com/13607045/118765076-e7aaeb00-b8bd-11eb-85de-6b624b5f1245.png">

**Reply by BeMySlaveDarlin** [reply]
![Снимок](https://user-images.githubusercontent.com/10584911/118784617-d69bb300-b898-11eb-8960-0c2571fa25a1.PNG)
As you see, barely 22MB with stubs and devtools loaded.

Can you paste gist with composer.json?

**Comment by BeMySlaveDarlin** [comment]
Installing via composer: You have to add `/YOUR_APP_PATH/vendor/bin` and `/YOUR_APP_PATH/vendor/phalcon/devtools` to you environment PATH (if using windows - search guide how to set up environments). Then you can use `phalcon` command.
This is common way to work with composer binaries and not about `phalcon-devtools` at all.
Also you can just symlink nessesary binaries as in git guide:
`ln -s /LOCAL_COMPOSER_DIR_OR_GLOBAL_COMPOSER_DIR/vendor/phalcon/devtools/phalcon /usr/local/bin/phalcon`


Installing via git: After you cloned repository, first of all in any cases you have to run `composer install` inside cloned repo directory.
After all you can do symlink and use `phalcon`,

**Reply by adgray** [reply]
Thank you for responding, but I was asking if there any up-to-date, reasonably complete instructions on how to install the Phalcon devtools, that **_don't_** assume I understand Composer. If you read my post, you can see that I did eventually get it working but at the cost of having 1 gigabyte of dependencies added, which is bizarre. This shouldn't be required just to run the devtools (and most likely isn't) but attempting to follow the poor instructions at the link I posted led to that.

**Reply by RickyJaff** [reply]
I also felt lost when I tried to use dev-tools for first time, but over time, you will find that instructions are fine, only thing is that it just takes time to learn and figure out things by yourself. Once you have the concept, you can do it easily , ofcourse if you have suggestion for the for improving docs, you can contribute.

**Reply by adgray** [reply]
In other words, once I understand how to install and use the tools, I'll understand how to install and use the tools :-)

I certainly hope to get some time to contribute to updated install docs for the devtools, once I'm more familiar with them. They shouldn't be hostile to new/less experienced users.
